core/services/job/spawner.go

```diff
@@ -0,0 +1,189 @@
+package job
+
+import (
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+)
+
+//go:generate mockery --name Spawner --output ./mocks/ --case=underscore
+
+type (
+	// The job spawner manages the spinning up and spinning down of the long-running
+	// services that perform the work described by job specs.  Each active job spec
+	// has 1 or more of these services associated with it.
+	//
+	// At present, Flux Monitor and Offchain Reporting jobs can only have a single
+	// "initiator", meaning that they only require a single service.  But the older
+	// "direct request" model allows for multiple initiators, which imply multiple
+	// services.
+	Spawner interface {
+		Start()
+		Stop()
+		CreateJob(spec Spec) error
+		DeleteJob(spec Spec) error
+		RegisterJobType(registration Registration)
+	}
+
+	spawner struct {
+		orm                    ORM
+		jobTypeRegistrations   map[Type]Registration
+		jobTypeRegistrationsMu sync.RWMutex
+		services               map[int32][]Service
+		chStopJob              chan int32
+		chStop                 chan struct{}
+		chDone                 chan struct{}
+	}
+
+	ServicesFactory func(spec Spec) ([]Service, error)
+
+	Registration struct {
+		JobType         Type
+		Spec            Spec
+		ServicesFactory ServicesFactory
+	}
+)
+
+var _ Spawner = (*spawner)(nil)
+
+func NewSpawner(orm ORM) *spawner {
+	return &spawner{
+		orm:                  orm,
+		jobTypeRegistrations: make(map[Type]Registration),
+		services:             make(map[int32][]Service),
+		chStopJob:            make(chan int32),
+		chStop:               make(chan struct{}),
+		chDone:               make(chan struct{}),
+	}
+}
+
+func (js *spawner) Start() {
+	go js.runLoop()
+}
+
+func (js *spawner) Stop() {
+	close(js.chStop)
+	<-js.chDone
+}
+
+func (js *spawner) RegisterJobType(registration Registration) {
+	js.jobTypeRegistrationsMu.Lock()
+	defer js.jobTypeRegistrationsMu.Unlock()
+
+	js.jobTypeRegistrations[registration.JobType] = registration
+}
+
+func (js *spawner) runLoop() {
+	defer close(js.chDone)
+
+	ticker := time.NewTicker(5 * time.Second)
+	defer ticker.Stop()
+
+	js.startUnclaimedServices()
+
+	for {
+		select {
+		case <-js.chStop:
+			js.stopAllServices()
+			return
+
+		case jobID := <-js.chStopJob:
+			js.stopService(jobID)
+
+		case <-ticker.C:
+			js.startUnclaimedServices()
+		}
+	}
+}
+
+func (js *spawner) startUnclaimedServices() {
+	jobSpecs, err := js.orm.UnclaimedJobs(js.jobTypeRegistrations, js.services)
```

Comment:
It doesn't yet -- I'm trying to cut scope, so right now, it assumes it's the only node in the cluster (since we don't support horizontal scaling yet in many other ways).

Reply:
Lets pls not do this. It will never get fixed and we just created even more work for ourselves.

We will get it done properly here.

Reply:
Or, worst case scenario, merge it in with a TODO